File: speakinsights-v3/backend/app/schemas/meeting.py
"""
SpeakInsights v3 — Meeting Schemas
"""

from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID
from typing import Optional


class MeetingCreate(BaseModel):
    """Schema for creating a new meeting."""
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    language: str = Field(default="auto", max_length=10)
    host_name: str = Field(..., min_length=1, max_length=100)


class MeetingResponse(BaseModel):
    """Schema for meeting response."""
    id: UUID
    title: str
    description: Optional[str] = None
    code: str
    language: str
    status: str
    host_name: str
    max_participants: int
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    participant_count: int = 0

    model_config = {"from_attributes": True}


class MeetingListResponse(BaseModel):
    """Schema for paginated meeting list."""
    meetings: list[MeetingResponse]
    total: int
    page: int = 1
    per_page: int = 20


class JoinRequest(BaseModel):
    """Schema for joining a meeting."""
    display_name: str = Field(..., min_length=1, max_length=100)


class JoinResponse(BaseModel):
    """Schema for join meeting response (after host approval)."""
    token: str
    room_id: str
    livekit_url: str
    participant_id: UUID


class ParticipantResponse(BaseModel):
    """Schema for participant response."""
    id: UUID
    display_name: str
    is_host: bool
    is_approved: bool
    is_active: bool
    joined_at: Optional[datetime] = None
    left_at: Optional[datetime] = None

    model_config = {"from_attributes": True}

File: speakinsights-v3/frontend/src/types/meeting.ts
export type MeetingStatus = 'waiting' | 'active' | 'processing' | 'completed' | 'cancelled';
export type ParticipantStatus = 'pending' | 'approved' | 'declined' | 'in-meeting' | 'left';

export interface Meeting {
  id: string;
  title: string;
  description?: string;
  code: string;
  language: string;
  status: MeetingStatus;
  host_name: string;
  created_at: string;
  started_at?: string;
  ended_at?: string;
  duration?: number;
  participant_count: number;
}

export interface Participant {
  id: string;
  meeting_id: string;
  name: string;
  status: ParticipantStatus;
  joined_at?: string;
  left_at?: string;
  is_host: boolean;
}

export interface CreateMeetingRequest {
  title: string;
  description?: string;
  language: string;
  host_name: string;
}

export interface CreateMeetingResponse {
  id: string;
  title: string;
  description?: string;
  code: string;
  language: string;
  status: string;
  host_name: string;
  max_participants: number;
  started_at?: string;
  ended_at?: string;
  created_at: string;
  updated_at?: string;
  participant_count: number;
}

export interface JoinRequest {
  name: string;
  meeting_code: string;
}

export interface JoinResponse {
  participant_id?: string;
  token?: string;
  room_id?: string;
  livekit_url?: string;
  status?: 'pending' | 'approved' | 'waiting';
  message?: string;
}

export interface LobbyParticipant {
  participant_id: string;
  name: string;
  requested_at: string;
}
